ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py

```diff
@@ -1416,6 +1417,45 @@ def test_paddle_frac(
     )
 
 
+# diagonal
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.diagonal",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=2,
+        max_num_dims=2,
```

Comment:
this seems a bit restrictive. `paddle.diagonal` can take any n-dim tensor as input. So ideally, you should be testing for the cases where `ndim > 2` as well 🙂